SS_prelim.tpl

```diff
@@ -1336,39 +1336,57 @@
   //  SS_Label_Info_6.8.4 #Call fxn get_natmort()
   echoinput << "ready to do natmort " << endl;
   get_natmort();
-  natM_M1 = natM; //  base M1 to which M2 is added
-
-  //  SS_Label_Info_6.8.5 #add M2 for predator mortality
-  if (N_pred > 0)
+  echoinput<<"ok"<<endl;
+  for (s = 1; s <= nseas; s++)
   {
-    for (f1 = 1; f1 <= N_pred; f1++)
+    t = t_base + s;
+  // set M equal to M1; M2 can be added later if predators are used
+    natM(t,0) = natM_M1(s);
+    for(p = 1; p <= pop; p++)
     {
-      pred_M2(f1, styr) = MGparm(predparm_pointer(f1)); //  base with no seasonal effect
-      if (nseas == 1)
-      {
-        for (gp = 1; gp <= N_GP * gender * N_settle_timings; gp++)
-        {
-          natM(1, gp) += pred_M2(f1, styr);
-        }
-      }
-      else
+      natM(t, p) = natM(t,0);
+    }
+
+  //  SS_Label_Info_6.8.5 #add M2 for predator mortality
+      if(N_pred>0)
       {
-        t = styr - 1; // resets for each predator
-        for (s = 1; s <= nseas; s++)
+        for (f1 = 1; f1 <= N_pred; f1++)
         {
-          t++;
-          for (gp = 1; gp <= N_GP * gender * N_settle_timings; gp++)
+          f = predator(f1);
+          pred_M2(f1, t) = mgp_adj(predparm_pointer(f1)); //  base with no seasonal effect
+          if (nseas > 1)
+            pred_M2(f1, t) *= mgp_adj(predparm_pointer(f1) + s);
+
+  //  a new array for indexing g and gpi could simplify below
+  //        for (gp = 1; gp <= N_GP * gender * N_settle_timings; gp++)
+
+          p = fleet_area(f);  //  area this predator occurs in
+          for (gp = 1; gp <= N_GP * gender; gp++)
           {
-            natM(s, gp) += pred_M2(f1, styr) * MGparm(predparm_pointer(f1) + s);
+            g = g_Start(gp); //  base platoon
+            for (settle = 1; settle <= N_settle_timings; settle++)
+            {
+              g += N_platoon;
+              int gpi = GP3(g); // GP*gender*settlement
+              natM(t, p,gpi) += pred_M2(f1, t) * sel_num(s, f, g);
+            }
           }
         }
       }
-    }
-  }
 
-  natM = value(natM);
+  echoinput<<" ready for surv "<<endl;
```

Comment:
If this is meant to help users, maybe a name more informative than surv would be helpful...I can't tell from the msg what is going on here.